Courses/Views/Screens/HomeView.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  HomeView.swift
+//  Courses
+//
+//  Created by Dmytro Churakov on 07.10.2022.
+//
+
+import SwiftUI
+
+struct HomeView: View {
+    @State var hasScrolled = false
+    
+    var body: some View {
+        ZStack {
+            Color("Background").ignoresSafeArea()
+            
+            ScrollView {
+                scrollDetection
+                featured
+                Color.clear.frame(height: 1000)
+            }
+            .coordinateSpace(name: "scroll")
+            .safeAreaInset(edge: .top, content: {
+                Color.clear.frame(height: 70)
+            })
+            .overlay(
+                NavigationBar(title: "Featured", hasScrolled: $hasScrolled)
+        )
+        }
+    }
+    
+    var scrollDetection: some View {
+        GeometryReader { proxy in
+            Color.clear.preference(key: ScrollPreferenceKey.self, value: proxy.frame(in: .named("scroll")).minY)
+        }
+        .frame(height: 0)
+        .onPreferenceChange(ScrollPreferenceKey.self, perform: { value in
+            withAnimation(.easeInOut) {
+                if value < 0 {
+                    hasScrolled = true
+                } else {
+                    hasScrolled = false
+                }
+            }
+            
+        }) 
+    }
+    
+    var featured: some View {
+        TabView {
+            ForEach(courses) { course in
+                GeometryReader { proxy in
+                    FeaturedItem(course: course)
+                        .padding(.vertical, 40)
+                        .rotation3DEffect(.degrees(proxy.frame(in: .global).minX / 5), axis: (x: 1, y: 1, z: 0))
+                        .shadow(color: Color("Shadow").opacity(0.4), radius: 10, x: 0, y: 10)
+                        .overlay(Image(course.image)
+                            .resizable()
+                            .aspectRatio(contentMode: .fit)
+                            .frame(height: 210)
+                            .offset(x: 40, y: -95)
+                    )
+                }
+            }
+        }
+        .tabViewStyle(.page(indexDisplayMode: .never))
+        .frame(height: 430)
+        .background(
+            Image("Blob 1")
```

Comment:
@DCHRKV what means `Blob`?

Reply:
@Den-Ree that's an image name from assets.